iphone/Maps/Classes/Editor/MWMEditorViewController.mm

```diff
@@ -624,18 +765,68 @@ - (void)cellSelect:(UITableViewCell *)cell
   MWMPlacePageCellType const cellType = [self cellTypeForIndexPath:indexPath];
   switch (cellType)
   {
-    case MWMPlacePageCellTypeStreet:
-      [self performSegueWithIdentifier:kStreetEditorSegue sender:nil];
+  case MWMPlacePageCellTypeStreet:
+    [self performSegueWithIdentifier:kStreetEditorSegue sender:nil];
+    break;
+  case MWMPlacePageCellTypeCuisine:
+    [self performSegueWithIdentifier:kCuisineEditorSegue sender:nil];
+    break;
+  case MWMPlacePageCellTypeCategory:
+    [self performSegueWithIdentifier:kCategoryEditorSegue sender:nil];
+    break;
+  case MWMPlacePageCellTypeReportButton:
+    switch (self.featureStatus)
+    {
+    case osm::Editor::FeatureStatus::Untouched:
+    {
+      __weak auto wself = self;
+      [self.alertController presentPlaceDoesntExistAlertWithBlock:^(NSString * additionalMessage)
+      {
+        __strong auto self = wself;
```

Comment:
Так любой код, который внутри блока будет просто использовать self, получается, невалиден?

> On 19.4.2016, at 14.22, Vlad Mihaylenko <notifications@github.com> wrote:
> 
> In iphone/Maps/Classes/Editor/MWMEditorViewController.mm:
> 
> > +    break;
> > +  case MWMPlacePageCellTypeCuisine:
> > +    [self performSegueWithIdentifier:kCuisineEditorSegue sender:nil];
> > +    break;
> > +  case MWMPlacePageCellTypeCategory:
> > +    [self performSegueWithIdentifier:kCategoryEditorSegue sender:nil];
> > +    break;
> > +  case MWMPlacePageCellTypeReportButton:
> > +    switch (self.featureStatus)
> > +    {
> > +    case osm::Editor::FeatureStatus::Untouched:
> > +    {
> > +      __weak auto wself = self;
> > +      [self.alertController presentPlaceDoesntExistAlertWithBlock:^(NSString * additionalMessage)
> > +      {
> > +        __strong auto self = wself;
> __weak нужен, чтобы ослабить сильную ссылку, потому что иначе блок ретейнит селф, а селф ретейнит блок, получается утечка памяти.
> __strong нужен для того, чтобы гарантировать, что слабая ссылка не умрет в процессе выполнения блока, и мы не получим ситуация, когда две строчки внутри блока выполнятся, а третья нет.
> self не уничтожится.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
>